sql/src/parser.rs

```diff
@@ -593,6 +579,78 @@ impl<'a> Parser<'a> {
         }))
     }
 
+    fn maybe_parse_and_check_key_partition(
+        &mut self,
+        columns: &[ColumnDef],
+    ) -> Result<Option<KeyPartition>> {
+        let linear = if self.consume_token("KEY") {
+            false
+        } else if self.consume_tokens(&["LINEAR", "KEY"]) {
+            true
+        } else {
+            return Ok(None);
+        };
+
+        let key_columns = self.parser.parse_parenthesized_column_list(Mandatory)?;
+
+        // Ensure at least one column for partition key.
+        if key_columns.is_empty() {
```

Comment:
In fact, the logic here is useless at present, because when the SQL is' KEY () ', `parser.parse_parenthesized_column_list` will directly throw an error. If you want to tell the user the exact cause of the error, you need to modify the parse logic.

Reply:
From the source code, it is yes. However, it can't be concluded from the signature from the `parse_parenthesized_column_list`. So I guess we should keep the check here.

Reply:
Maybe we should leave a todo here to record this problem. Later, we can consider using our own `parse_parenthesized_column_list` to replace the native method of `sqlparser`.

Reply:
I see. What you are afraid of is the strange error reported to user. I will add some customized error here.